experiment/simulation/js/main.js

```diff
@@ -1 +1,478 @@
+                overlays: [
+                    { type: "Label", options: { label: "B0", id: "b0", location: [0.3, 1.7] } }
+                ],
+            })
+        );
+        // carry input
+        gate.addInputPoints(
+            jsPlumbInstance.addEndpoint(element, {
+                anchor: [1, 0.5, 1, 0, 7, 0],
+                source: true,
+                target: true,
+                connectionsDetachable: false,
+                uuid: "input:2:" + id,
+                overlays: [
+                    { type: "Label", options: { label: "Cin", id: "cin", location: [-1, 0.2] } }
+                ],
+            })
+        );
+    }
+}
+
+// Initialise Half adder experiment by generating and adding gates and components to the circuit board at given positions
+export function initHalfAdder() {
+    let ids = ["Input-0", "Input-1", "Output-2", "Output-3"]; // [A B Sum Carry Out]
+    let types = ["Input", "Input", "Output", "Output"];
+    let names = ["A", "B", "Sum", "Carry"];
+    let positions = [
+        { x: 40, y: 200 },
+        { x: 40, y: 550 },
+        { x: 820, y: 200 },
+        { x: 820, y: 550 },
+    ];
+    for (let i = 0; i < ids.length; i++) {
+        let gate = new gatejs.Gate(types[i]);
+        gate.setId(ids[i]);
+        gate.setName(names[i]);
+        const component = gate.generateComponent();
+        const parent = document.getElementById("working-area");
+        parent.insertAdjacentHTML('beforeend', component);
+        gate.registerComponent("working-area",positions[i].x, positions[i].y);;
+    }
+}
+
+// Initialise Full adder experiment by generating and adding gates and components to the circuit board at given positions
+export function initFullAdder() {
+    let ids = ["Input-0", "Input-1", "Input-2", "Output-3", "Output-4"]; // [A,B,carry -input,Sum,carry-output]
+    let types = ["Input", "Input", "Input", "Output", "Output"];
+    let names = ["A", "B", "CarryIn", "Sum", "CarryOut"];
+    let positions = [
+        { x: 40, y: 150 },
+        { x: 40, y: 375 },
+        { x: 40, y: 600 },
+        { x: 820, y: 262.5 },
+        { x: 820, y: 487.5 },
+    ];
+    for (let i = 0; i < ids.length; i++) {
+        let gate = new gatejs.Gate(types[i]);
+        gate.setId(ids[i]);
+        gate.setName(names[i]);
+        const component = gate.generateComponent();
+        const parent = document.getElementById("working-area");
+        parent.insertAdjacentHTML('beforeend', component);
+        gate.registerComponent("working-area",positions[i].x, positions[i].y);
+    }
+}
+
+// Initialise Ripple carry adder experiment by generating and adding gates and components to the circuit board at given positions
+export function initRippleAdder() {
+    let ids = ["Input-0", "Input-1", "Output-2", "Input-3", "Input-4", "Output-5", "Input-6", "Input-7", "Output-8", "Input-9", "Input-10", "Output-11", "Output-12", "Input-13"] // [A0,B0,Sum0,A1,B1,Sum1,A2,B2,Sum2,A3,B3,Sum3,CarryOut, CarryIn]

```

Comment:
We have still used capitalised letters as they followed a particular format of ID which we don't want to change as it will break the code and even clash with a lot of other variables with the same names. That's why we had to pick up this format.